django/core/serializers/python.py

```diff
@@ -75,42 +75,64 @@ def getvalue(self):
         return self.objects
 
 
-def Deserializer(object_list, *, using=DEFAULT_DB_ALIAS, ignorenonexistent=False, **options):
+class Deserializer(base.Deserializer):
     """
     Deserialize simple Python objects back into Django ORM instances.
-
     It's expected that you pass the Python objects themselves (instead of a
     stream or a string) to the constructor
     """
-    handle_forward_references = options.pop('handle_forward_references', False)
-    field_names_cache = {}  # Model: <list of field_names>
-
-    for d in object_list:
+    def __init__(self, object_list, *, using=DEFAULT_DB_ALIAS, ignorenonexistent=False, **options):
+        super().__init__(stream_or_string=object_list, **options)
+        self.handle_forward_references = options.pop(
+            'handle_forward_references', False)
+        self.using = using
+        self.ignore = ignorenonexistent
+
+    def __iter__(self):
+        for obj in self.stream:
+            yield self._handle_object(obj)
+
+    def _handle_m2m_fields(self, field, field_value):
+        values = base.deserialize_m2m_values(field, field_value,
+                                             self.using,
+                                             self.handle_forward_references)
+        return values
+
+    def _handle_foreign_key_field(self, field, field_value):
+        value = base.deserialize_fk_value(field, field_value,
+                                          self.using,
+                                          self.handle_forward_references)
+        return value
+
+    def _handle_object(self, model_obj):
+        data = {}
+        m2m_data = {}
+        deferred_fields = {}
+        field_names_cache = {}  # Model: <list of field_names>
         # Look up the model and starting build a dict of data for it.
         try:
-            Model = _get_model(d["model"])
+            Model = self._get_model(model_obj["model"])
         except base.DeserializationError:
-            if ignorenonexistent:
-                continue
+            if self.ignore:
+                return None
```

Comment:
returning `None` isn't perfectly equivalent to `continue`. I think it might be cleaner to move some or all of `_handle_object` back into `__iter__`. This allows us to use `continue` again.

Reply:
I thought `return None` will break the method call and pick the next object.